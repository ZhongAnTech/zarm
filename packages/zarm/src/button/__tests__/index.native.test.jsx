import { TouchableHighlight, Text } from 'react-native';
import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../index.native';

describe('Button', () => {
  it('renders correctly', () => {
    const wrapper = render(<Button>foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('theme', () => {
    const wrapper = render(<Button theme="primary">foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('block', () => {
    const wrapper = render(<Button block>foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('ghost', () => {
    const wrapper = render(<Button ghost>foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('size', () => {
    const wrapper = render(<Button size="lg">foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('shape is radius', () => {
    const wrapper = render(<Button shape="radius">foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('children is valid element', () => {
    const wrapper = render(
      <Button>
        <Text>foo</Text>
      </Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('icon', () => {
    const wrapper = render(
      <Button icon={<img alt="" src="https://zarm.design/images/logo.ce68565d.svg" />}>foo</Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('loading', () => {
    const wrapper = render(<Button loading>foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('disabled', () => {
    const wrapper = render(<Button disabled>foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('onClick', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Button onClick={onClick}>foo</Button>);
    wrapper.find(TouchableHighlight).props().onPress();
    expect(onClick).toBeCalled();
  });

  it('pressIn', () => {
    const wrapper = shallow(<Button>foo</Button>);
    wrapper.find(TouchableHighlight).props().onPressIn();
    expect(wrapper.state('isActive')).toBe(true);
  });

  it('pressOut', () => {
    const wrapper = shallow(<Button>foo</Button>);
    wrapper.find(TouchableHighlight).props().onPressOut();
    expect(wrapper.state('isActive')).toBe(false);
  });

  it('ghost active', () => {
    const wrapper = shallow(<Button ghost>foo</Button>);
    wrapper.setState({ isActive: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('ghost disabled', () => {
    const wrapper = shallow(
      <Button ghost disabled>
        foo
      </Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('circle size', () => {
    const wrapper = render(
      <Button shape="circle" size="lg">
        foo
      </Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
