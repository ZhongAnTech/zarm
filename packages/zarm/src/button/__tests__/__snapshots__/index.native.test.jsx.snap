// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button block 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      foo
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`Button children is valid element 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      foo
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`Button circle size 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      foo
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`Button disabled 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      foo
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`Button ghost 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      foo
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`Button ghost active 1`] = `
<TouchableOpacity
  activeOpacity={1}
  block={false}
  disabled={false}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 44,
        "justifyContent": "center",
        "paddingLeft": 16,
        "paddingRight": 16,
      },
      Object {
        "height": 44,
        "paddingLeft": 16,
        "paddingRight": 16,
      },
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e6e6e6",
        "borderWidth": 1,
      },
      Object {
        "borderRadius": 4,
      },
      Object {
        "backgroundColor": "#e6e6e6",
        "borderColor": "#e6e6e6",
      },
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 1,
      },
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#fff",
      },
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#e6e6e6",
      },
      false,
      false,
      false,
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
          },
          Object {
            "fontSize": 16,
          },
          Object {
            "color": "#464646",
          },
          Object {
            "color": "#464646",
          },
          false,
          Object {
            "color": "#fff",
          },
          Object {
            "color": "#e6e6e6",
          },
          false,
        ]
      }
    >
      foo
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button ghost disabled 1`] = `
<TouchableOpacity
  activeOpacity={1}
  block={false}
  disabled={true}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 44,
        "justifyContent": "center",
        "paddingLeft": 16,
        "paddingRight": 16,
      },
      Object {
        "height": 44,
        "paddingLeft": 16,
        "paddingRight": 16,
      },
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e6e6e6",
        "borderWidth": 1,
      },
      Object {
        "borderRadius": 4,
      },
      false,
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 1,
      },
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#fff",
      },
      false,
      Object {
        "opacity": 0.5,
      },
      Object {
        "borderColor": "#bcbcbc",
        "opacity": 1,
      },
      false,
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
          },
          Object {
            "fontSize": 16,
          },
          Object {
            "color": "#464646",
          },
          false,
          Object {
            "color": "#bcbcbc",
          },
          Object {
            "color": "#fff",
          },
          false,
          Object {
            "color": "#bcbcbc",
          },
        ]
      }
    >
      foo
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button icon 1`] = `
<react-native-mock>
  <react-native-mock>
    <img
      alt=""
      src="https://zarm.design/images/logo.ce68565d.svg"
    />
    <react-native-mock>
      foo
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`Button loading 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock />
    <react-native-mock>
      foo
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`Button renders correctly 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      foo
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`Button shape is radius 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      foo
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`Button size 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      foo
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`Button theme 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      foo
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;
